src/Compilers/Core/Portable/InternalUtilities/FailFast.cs

```diff
@@ -34,10 +34,35 @@ internal static void OnFatalException(Exception exception)
             }
 #endif
 
+            DumpStackTrace(exception);
+
             Environment.FailFast(exception.ToString(), exception);
             throw ExceptionUtilities.Unreachable; // to satisfy [DoesNotReturn]
         }
 
+        /// <summary>
+        /// Dumps the stack trace of the exception and the handler to the console. This is useful
+        /// for debugging unit tests that hit a fatal exception
+        /// </summary>
+        [Conditional("DEBUG")]
+        private static void DumpStackTrace(Exception exception)
+        {
+            Console.WriteLine("Dumping info before call to failfast");
```

Comment:
Is it worth writing this to standard error instead of standard out if we ever have something redirecting the latter? My suggestion could be a bad idea if we have tools that don't handle standard error well...

Reply:
Pretty much everything in our tool chain uses standard out even for errors. That's why I chose it here. Worried standard out would create yet another diagnostics wrinkle 😦